src/cache/cache.h

```diff
@@ -0,0 +1,79 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#ifndef CACHE_H
+#define CACHE_H
+
+#include "./cache_includes.h"
+#include "./lru_cache_manager/lru_cache_manager.h"
+#include "./rax_cache_storage/rax_cache_storage.h"
+#include "xxhash_query_hash/xxhash_query_hash.h"
+
+/**
+ * A struct to for read queries cache storage. 
+ * Holds a managment struct (LRU policy) and a RAX radix tree for storage.
+ * A read-write lock is set for multi-threaded operatins.
+ */
+typedef struct Cache
+{
+    LRUCacheManager *lruCacheManager; // Cache Manager - LRU policy
+    RaxCacheStorage *raxCacheStorage; // Storage - Rax radix tree
```

Comment:
Incase RaxCacheStorage is simply a wrapper around RAX then let's remove this wrapping and use RAX directly.

Reply:
done